templates/dev.rake

```diff
@@ -1,4 +1,4 @@
-if Rails.env.development? || Rails.env.test?
+unless Rails.env.production?
```

Comment:
But... why? We want these environments to be handled differently. We already mapped the entire production config to staging for dev/prod parity.

Now we have a custom configuration throughout the app running complex checks instead of just using the Rails convention that was already designed for this exact purpose.

In particular, for this line, we're going to have to use something nasty like:
```ruby
if Rails.env.development? or Rails.env.test? or ENV.fetch("HEROKU_APP_NAME", "").include?("staging-pr-")
  ...
```

While part of me may be resisting change.. the main part just doesn't understand -- despite the Heroku recommendations.

In this same vein, should we get rid of the `test` environment because it's similar to `development`? Why even have `development`? Why not just run everything in `RAILS_ENV=production`